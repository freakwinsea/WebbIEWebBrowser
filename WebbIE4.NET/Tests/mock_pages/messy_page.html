<!DOCTYPE html>
<html>
<head>
    <title>Messy Page Test</title>
</head>
<body>
    <h1>Welcome</p> <!-- Mismatched tag -->

    <div>
        <p>Here is a link without an href.</p>
        <a>I am not a link.</a>
    </div>

    <!-- Unlabeled form controls -->
    <form>
        <input type="text" name="first_name" placeholder="First Name">
        <br>
        <input type="submit" value="Submit Here">
    </form>

    <div>
        <p>A link with no text content: <a href="/empty"></a></p>
    </div>

    <button>
        <p>A button with a paragraph inside</p>
    </button>

    <div style="display: none;">
        <p>This text is invisible and should not be parsed.</p>
    </div>

</body>
</html>
