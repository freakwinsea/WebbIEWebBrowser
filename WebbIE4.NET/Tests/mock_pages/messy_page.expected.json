[
  {
    "id": "webbie-id-0",
    "type": "h1",
    "text": "Welcome",
    "href": ""
  },
  {
    "id": "webbie-id-1",
    "type": "p",
    "text": "Here is a link without an href.",
    "href": ""
  },
  {
    "id": "webbie-id-2",
    "type": "text",
    "text": "I am not a link.",
    "href": ""
  },
  {
    "id": "webbie-id-3",
    "type": "text",
    "text": "First Name",
    "href": ""
  },
  {
    "id": "webbie-id-4",
    "type": "submit",
    "text": "Submit Here",
    "href": ""
  },
  {
    "id": "webbie-id-5",
    "type": "p",
    "text": "A link with no text content:",
    "href": ""
  },
  {
    "id": "webbie-id-6",
    "type": "button",
    "text": "A button with a paragraph inside",
    "href": ""
  }
]
